src/resources/rmt_esp32.cc

```diff
@@ -31,17 +31,74 @@ namespace toit {
 
 const rmt_channel_t kInvalidChannel = static_cast<rmt_channel_t>(-1);
 
+// Copied from rmt_legacy.c.
+#define RMT_RX_CHANNEL_ENCODING_START (SOC_RMT_CHANNELS_PER_GROUP-SOC_RMT_TX_CANDIDATES_PER_GROUP)
+#define RMT_TX_CHANNEL_ENCODING_END   (SOC_RMT_TX_CANDIDATES_PER_GROUP-1)
+
+// Copied from rmt_legacy.c.
+#define RMT_IS_RX_CHANNEL(channel) ((channel) >= RMT_RX_CHANNEL_ENCODING_START)
+#define RMT_IS_TX_CHANNEL(channel) ((channel) <= RMT_TX_CHANNEL_ENCODING_END)
+
 #ifndef SOC_RMT_CHANNELS_PER_GROUP
 #error "SOC_RMT_CHANNELS_PER_GROUP not defined"
 #endif
 
-ResourcePool<rmt_channel_t, kInvalidChannel> rmt_channels(
-    RMT_CHANNEL_0, RMT_CHANNEL_1, RMT_CHANNEL_2, RMT_CHANNEL_3
+#if SOC_RMT_CHANNELS_PER_GROUP == SOC_RMT_TX_CANDIDATES_PER_GROUP
+// All channels are TX channels. We assume that they also are RX channels.
+// This is the setup on the ESP32 and ESP32S2 chips.
+ResourcePool<rmt_channel_t, kInvalidChannel> rmt_channels_(
```

Comment:
Should these be static (pools and pointers)?

Reply:
done.